tests/test_system_profile.py

```diff
@@ -387,3 +387,47 @@ def test_system_profile_operating_system(mq_create_or_update_host, api_get):
         item_key = f"{item['value']['name']}_{item['value']['major']}.{item['value']['minor']}"
         item_count = item["count"]
         assert item_count == os_dict[item_key]["count"]
+
+
+def test_system_profile_sap_system(mq_create_or_update_host, api_get):
+    # Create some sap systems
+    ordered_sap_system_data = [True, True, False, False, True, False]
+    ordered_insights_ids = [generate_uuid() for _ in range(len(ordered_sap_system_data))]
+
+    # Create an association between the insights IDs
+    ordered_host_data = dict(zip(ordered_insights_ids, ordered_sap_system_data))
+
+    # Create a shuffled list of insights_ids so we can create the hosts in a random order
+    shuffled_insights_ids = ordered_insights_ids.copy()
+    random.shuffle(shuffled_insights_ids)
+
+    # Create hosts for the above host data (in shuffled order)
+    _ = [
+        mq_create_or_update_host(
+            minimal_host(insights_id=insights_id, system_profile={"sap_system": ordered_host_data[insights_id]})
+        )
+        for insights_id in shuffled_insights_ids
+    ]
+    url = build_system_profile_sap_system_url()
+
+    sap_list = []
+    not_sap_list = []
+    for datum in ordered_sap_system_data:
+        if datum:
+            sap_list.append(datum)
+        else:
+            not_sap_list.append(datum)
+
+    with patch("api.system_profile.get_flag_value", return_value=True):
+        # Validate the basics, i.e. response code and results size
+        response_status, response_data = api_get(url)
+        assert response_status == 200
+        assert len(list(set(ordered_sap_system_data))) == len(response_data["results"])
```

Comment:
Does it need to be converted to a list?
```suggestion
        assert len(set(ordered_sap_system_data)) == len(response_data["results"])
```

Reply:
ok, pushed